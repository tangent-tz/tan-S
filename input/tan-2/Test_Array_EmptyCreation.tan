main{
    print 45644 \n; 
    
    var x1 := new [int] (2*5 - 2);
    print x1 \n; 
    
    var x2 := new [float] (2*5 - 2);
    print x2 \n;
    
    var x3 := new [string] (2*5 - 2);
    print x3 \n;
    
    var x4 := new [char] (2*5 - 2);
    print x4 \n;
    
    var x5 := new [bool] (2*5 - 2);
    print x5 \n;
    
    print 123 \n;
    
    
    #=====================================================================
    #TEST MULTI-DIMENSIONAL ARRAYS
    var ints2d := new [[int]](3); 
    print ints2d \n; 
    print new [[int]](3) \n\n; 
    
    
    var floats3d := new [[[float]]](2); 
    print floats3d \n; 
    print new [[[float]]](2) \n\n; 
    
    
    var strings4d := new [[[[string]]]](2); 
    print strings4d \n; 
    print new [[[[string]]]](2) \n\n;
    
    
    var chars3d := new [[[char]]](3); 
    print chars3d \n; 
    print new [[[char]]](3) \n\n;
    
    
    var bools3d := new [[[bool]]](2); 
    print bools3d \n; 
    print new [[[bool]]](2) \n\n;    
    
}