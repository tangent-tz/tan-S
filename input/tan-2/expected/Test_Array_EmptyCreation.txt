45644
[ 0, 0, 0, 0, 0, 0, 0, 0 ]
[ 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000 ]
[ , , , , , , ,  ]
[  ,  ,  ,  ,  ,  ,  ,   ]
[ false, false, false, false, false, false, false, false ]
123
[ q, q, q ]
[ q, q, q ]

[ q, q ]
[ q, q ]

[ q, q ]
[ q, q ]

[ q, q, q ]
[ q, q, q ]

[ q, q ]
[ q, q ]

